azurerm/resource_arm_front_door.go

```diff
@@ -0,0 +1,1459 @@
+	if len(input) == 0 {
+		return &[]frontdoor.LoadBalancingSettingsModel{}
+	}
+
+	output := make([]frontdoor.LoadBalancingSettingsModel, 0)
+
+	for _, lbs := range input {
+		loadBalanceSetting := lbs.(map[string]interface{})
+
+		name := loadBalanceSetting["name"].(string)
+		sampleSize := int32(loadBalanceSetting["sample_size"].(int))
+		successfulSamplesRequired := int32(loadBalanceSetting["successful_samples_required"].(int))
+		additionalLatencyMilliseconds := int32(loadBalanceSetting["additional_latency_milliseconds"].(int))
+		id := utils.String(helper.GetFrontDoorSubResourceId(subscriptionId, resourceGroup, frontDoorName, "LoadBalancingSettings", name))
+
+		result := frontdoor.LoadBalancingSettingsModel{
+			ID:   id,
+			Name: utils.String(name),
+			LoadBalancingSettingsProperties: &frontdoor.LoadBalancingSettingsProperties{
+				SampleSize:                    utils.Int32(sampleSize),
+				SuccessfulSamplesRequired:     utils.Int32(successfulSamplesRequired),
+				AdditionalLatencyMilliseconds: utils.Int32(additionalLatencyMilliseconds),
+			},
+		}
+
+		output = append(output, result)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorRoutingRule(input []interface{}, subscriptionId string, resourceGroup, frontDoorName string) *[]frontdoor.RoutingRule {
+	if len(input) == 0 {
+		return nil
+	}
+
+	output := make([]frontdoor.RoutingRule, 0)
+
+	for _, rr := range input {
+		routingRule := rr.(map[string]interface{})
+
+		id := routingRule["id"].(string)
+		frontendEndpoints := routingRule["frontend_endpoints"].([]interface{})
+		acceptedProtocols := routingRule["accepted_protocols"].([]interface{})
+		ptm := routingRule["patterns_to_match"].([]interface{})
+		enabled := routingRule["enabled"].(bool)
+		name := routingRule["name"].(string)
+
+		patternsToMatch := make([]string, 0)
+
+		for _, p := range ptm {
+			patternsToMatch = append(patternsToMatch, p.(string))
+		}
+
+		var routingConfiguration frontdoor.BasicRouteConfiguration
+
+		if rc := routingRule["redirect_configuration"].([]interface{}); len(rc) != 0 {
+			routingConfiguration = expandArmFrontDoorRedirectConfiguration(rc)
+		} else if fc := routingRule["forwarding_configuration"].([]interface{}); len(fc) != 0 {
+			routingConfiguration = expandArmFrontDoorForwardingConfiguration(fc, subscriptionId, resourceGroup, frontDoorName)
+		}
+
+		currentRoutingRule := frontdoor.RoutingRule{
+			ID:   utils.String(id),
+			Name: utils.String(name),
+			RoutingRuleProperties: &frontdoor.RoutingRuleProperties{
+				//ResourceState:
+				FrontendEndpoints:  expandArmFrontDoorFrontEndEndpoints(frontendEndpoints, subscriptionId, resourceGroup, frontDoorName),
+				AcceptedProtocols:  expandArmFrontDoorAcceptedProtocols(acceptedProtocols),
+				PatternsToMatch:    &patternsToMatch,
+				EnabledState:       frontdoor.RoutingRuleEnabledState(expandArmFrontDoorEnabledState(enabled)),
```

Comment:
Could this be:

```suggestion
				EnabledState:       expandArmFrontDoorEnabledState(enabled),
```

Reply:
I don't think so because they exposed `RoutingRuleEnabledState` as it's own type and it shares the values with the `EnabledState` type. This is how I convert from `bool` to the various other `EnabledState`s.
```
type RoutingRuleEnabledState string
```
```
type EnabledState string
```